### PR TITLE
Added getter_index attribute to parameter model

### PR DESCRIPTION
**Pull request by epcsamuli**

### CODE REVIEW

**Review by epcAnker — DISMISSED** [review]
to get rid of CI errors: Need to run black formatter.

Also (i just had this issue myself) may need to follow step 4 & 5 of https://epcpower.atlassian.net/wiki/spaces/EN/pages/44892161/How+to+create+a+new+Parameter+Attribute.

File: epyqlib/pm/parametermodel.py [hunk 1]
```diff
@@ -1696,6 +1705,7 @@ def merge(name, *types):
     merge("internal_type", Parameter),
     merge("internal_scale_factor", Parameter),
     merge("reject_from_inactive_interfaces", Parameter),
+    merge("getter_index", Parameter),
```
[the remark this answers is not in the document]
Comment (epcsamuli) [reply epyqlib/pm/parametermodel.py]: The value goes for the `getter_function`, I moved it below that

[the remark this answers is not in the document]
Comment (altendky) [reply epyqlib/pm/parametermodel.py]: Is there a reason that the index feature should not be made to work with both getting and setting?  Or maybe the plan is just taking it one step at a time.

As Alex noted above, the test data file needs to be updated.  The failing test is checking that the data is still serialized to the file as intended, but adding this attribute changes the serialization.  I generally copy and paste the new output in the test into the file it is being checked against and then do a diff with git and confirm that the changes are what I intended.  If so, those changes to the reference file can be committed.

[the remark this answers is not in the document]
Comment (epcsamuli) [reply epyqlib/pm/parametermodel.py]: Well there's no particular reason why it couldn't be extended to cover setting as well, but the feature specification only considered read-only values, so indexed setters were not implemented yet.

[the remark this answers is not in the document]
Comment (epcsamuli) [reply epyqlib/pm/parametermodel.py]: I tried the copy-pasting approach and it seemed to work, thanks!

**Review by altendky — APPROVED** [review]

**Pull request merged**